### DISCUSSION TITLE
Draft User Workflow

### DISCUSSION THREAD

**Discussion by clizbe** [TulipaEnergy/TulipaEnergyModel.jl#700] [Ideas]
**Prereqs:**
- Download Julia
- Add Tulipa
- ~Get an ESDL MapEditor account~
- Download VSCode (optional)
- Add Pluto (optional)

**Create 3 Core files:**
1) Graph file
    - What are the requirements?
    - Where should this be explained?
    - ~Use ESDL MapEditor~
    - ~Include assigning profiles?~
	~Export ESDL to analysis folder~

2) Scenario file
    - How?
    - How is this related to editing data in TIO?

3) Partitions file
    - How?

**Follow Pipeline**
Create a Julia script or Pluto notebook (recommended) to follow:
https://github.com/TulipaEnergy/pipeline-example

Summary of pipeline:
1) Use TIO to read files
1a) Edit data for alternatives?
#414
2) Send data to TC
3) Send data to TEM
    - Future: With feature specifications (see https://github.com/TulipaEnergy/TulipaEnergyModel.jl/issues/461#issuecomment-2242797692)
4) Graph results
    - Currently: Hand-craft as needed (in Julia!)
    - Future: Import data into Pluto graph template

Related:
- #415 
- #414 
- #289
- #499
- Maybe some ideas in here while we were talking ESDL: #56

**Comment by clizbe** [comment]
Just wanted to write this down for my own sanity. Also to see if it makes sense and/or can be simplified.

**Comment by clizbe** [comment]
Random thought:  Another model I know has a files that explicitly state whether a specific technology is allowed (or prohibited) in an area. This is specified _in addition_ to their normal method of allowing tech in areas, as an _override switch_ to be sure those specific techs are allowed/prohibited.
Long story short this shows that model does not have oversight of all the data going into the model for a given scenario. So some kind of "overview" and/or sanity checks of a scenario would be nice.

**Comment by datejada** [comment]
@IsaiMaganTNO, we can help to add more things to this workflow from our experience in the OBZ case study. So, let's keep this discussion on our radar and update it with the main findings.

**Reply by clizbe** [reply]
Yes and feel free to update the main block of the discussion above with the new workflow. Delete what's there and put what you actually have. :)

**Comment by clizbe** [comment]
Closing this discussion as we've moved on. It's linked in #974 for background info.
